src/app/georeferencer/qgsgeorefmainwindow.cpp

```diff
@@ -296,6 +305,94 @@ void QgsGeoreferencerMainWindow::openRaster()
   mWorldFileName = guessWorldFileName( mRasterFileName );
 }
 
+void QgsGeoreferencerMainWindow::dropEvent( QDropEvent *event )
+{
+  // dragging app is locked for the duration of dropEvent. This causes explorer windows to hang
+  // while large projects/layers are loaded. So instead we return from dropEvent as quickly as possible
+  // and do the actual handling of the drop after a very short timeout
+  QTimer *timer = new QTimer( this );
+  timer->setSingleShot( true );
+  timer->setInterval( 50 );
+
+  // get the file list
+  QList<QUrl>::iterator i;
+  QList<QUrl>urls = event->mimeData()->urls();
+  QString file;
+  for ( i = urls.begin(); i != urls.end(); ++i )
+  {
+    QString fileName = i->toLocalFile();
+#ifdef Q_OS_MAC
```

Comment:
Did you copy this from somewhere?
Should we put this into a utils file instead?

Reply:
I took it from the layout designer (I assume qgisapp has the same code). +1 for code deduplication, but I'm not sure where we'd put this static function.

Reply:
QgsGuiUtils?

Reply:
@m-kuhn , actually, the underlying issue was fixed in Qt 5.4.1 (https://codereview.qt-project.org/c/qt/qtbase/+/103158) -- are we planning to support Qt5 pre 5.4.1 for QGIS 3.18? If not, I'd suggest we get rid of this hack altogether.

Reply:
https://github.com/qgis/QGIS-Enhancement-Proposals/issues/197
+1 to get rid of it

Reply:
@m-kuhn , done.